File: test/unit-testing/arc.c
#define DWG_TYPE DWG_TYPE_ARC
#include "common.c"

void
api_process (dwg_object *obj)
{
  int error;
  double radius, thickness, start_angle, end_angle;
  dwg_point_3d center, extrusion;

  dwg_ent_arc *arc = dwg_object_to_ARC (obj);

  CHK_ENTITY_TYPE_W_OLD (arc, ARC, radius, BD);
  CHK_ENTITY_TYPE_W_OLD (arc, ARC, thickness, BD);
  CHK_ENTITY_3RD_W_OLD (arc, ARC, extrusion);
  CHK_ENTITY_3RD_W_OLD (arc, ARC, center);
  CHK_ENTITY_TYPE_W_OLD (arc, ARC, start_angle, BD);
  CHK_ENTITY_MAX (arc, ARC, start_angle, BD, MAX_ANGLE);
  CHK_ENTITY_TYPE_W_OLD (arc, ARC, end_angle, BD);
  CHK_ENTITY_MAX (arc, ARC, end_angle, BD, MAX_ANGLE);
}

File: bindings/javascript/src/types/entities/arc.ts
import { DwgPoint3D } from '../common'
import { DwgEntity } from './entity'

export interface DwgArcEntity extends DwgEntity {
  type: 'ARC'
  thickness: number
  center: DwgPoint3D
  radius: number
  startAngle: number
  endAngle: number
  extrusionDirection: DwgPoint3D
}
